pkg/common/providers/mock/mock.go

```diff
@@ -66,27 +78,108 @@ func (m *MockProvider) GetCluster(clusterID string) (*spi.Cluster, error) {
+
+	return []byte(kubeconfig), nil
 }
 
 // CheckQuota mocks a check quota operation.
 func (m *MockProvider) CheckQuota() (bool, error) {
-	return false, fmt.Errorf("check quota is currently unsupported by the mock provider")
+	if m.env == "fail" {
+		return false, fmt.Errorf("failed to get versions: Some fake error")
+	}
+
+	// By default this will pass.
+	// If you want a purposeful CheckQuota failure, you should set up a `fail` environment
+	return true, nil
 }
 
 // InstallAddons mocks an install addons operation.
 func (m *MockProvider) InstallAddons(clusterID string, addonIDs []string) (int, error) {
-	return 0, fmt.Errorf("install addons is currently unsupported by the mock provider")
+	if clusterID == "fail" {
+		return 0, fmt.Errorf("failed to get versions: Some fake error")
+	}
+
+	cluster, err := m.GetCluster(clusterID)
+	if err != nil {
+		return 0, fmt.Errorf("Unable to retrieve cluster: %s", err.Error())
+	}
+	// We can't access the addons field directly so we have to rebuild the cluster object from scratch
+	// This is fine as any real provider would call an external API to update or retrieve addons and
+	// we lose no state doing this.
+	m.clusters[clusterID] = spi.NewClusterBuilder().
+		ID(clusterID).
+		Name(cluster.Name()).
+		Version(cluster.Version()).
+		State(cluster.State()).
+		CloudProvider(cluster.CloudProvider()).
+		Region(cluster.Region()).
+		ExpirationTimestamp(cluster.ExpirationTimestamp()).
+		Flavour(cluster.Flavour()).
+		Addons(addonIDs).
+		Build()
+
+	return len(addonIDs), nil
 }
 
 // Versions mocks a versions operation.
 func (m *MockProvider) Versions() (*spi.VersionList, error) {
-	return nil, fmt.Errorf("versions is currently unsupported by the mock provider")
+	if m.env == "fail" {
+		return nil, fmt.Errorf("Fake error returning version list")
+	}
+	versions := []*spi.Version{
```

Comment:
I feel like we should make a public helper function here so that tests can populate their own version lists -- this should be fine, as tests will likely be using the MockProvider directly.

Reply:
I agree. We may want to stress version selection logic in novel ways so we should be able to set different use cases. I'll update it now.